### ISSUE TITLE
What is "<path_teacher_pkl>"?

### ISSUE THREAD

**Issue by jsuit** [adri-romsor/FitNets#1]
In cifar10_fitnet19_all.yaml:

cost: !obj:costs.KDCost.KDCost {
          teacher_path: "<path_teacher_pkl>",
          temperature: 3,
          lambda_teach: 4,
          hints: [[10,1]],
        },

what is the teacher_path supposed to be? When run, I get "No such file or directory: '<path_teacher_pkl>' ".

Also, What is the difference between cifar10_fitnet19_all.yaml and cifar10_fitnet19_valid.yam? What does the valid and all stand for?

**Comment by adri-romsor** [comment]
- teacher_path is the path to the .pkl of the teacher model. The teacher model is the maxout network in [a]. You have to train the teacher first. If you are using pylearn2, you can find the yaml file of the teacher in /pylearn2/pylearn2/scripts/papers/maxout/. After training the teacher, set the path to the resulting pkl as teacher_path.

[a] Goodfellow, I.J., Warde-Farley, D., Mirza, M., Courville, A., and Bengio, Y. Maxout networks. In ICML, 2013.
- cifar10_fitnet19_valid.yaml divides the training set into a training set and a validation set. This yaml is used to determine the number of epochs to train the FitNet. It will give you the number of epochs to train the first and second stages. After that, you can use cifar_10_fitnet19_all.yaml to retrain on the whole training set. Set the number of epochs of the 2nd stage here:
  
  termination_criterion: !obj:pylearn2.termination_criteria.EpochCounter {
            max_epochs: XXXX,
  },
  and set the number of epochs of the 1st stage as an optional parameter when running fitnets_training.py: -he YYYY.
